services/brig/test/integration/API/OAuth.hs

```diff
@@ -0,0 +1,865 @@
+  let refreshAccessTokenRequest = OAuthRefreshAccessTokenRequest OAuthGrantTypeRefreshToken cid secret (oatRefreshToken accessToken)
+  refreshedToken <- refreshOAuthAccessToken brig refreshAccessTokenRequest
+  get (nginz . paths ["self"] . authHeader (oatAccessToken refreshedToken)) !!! const 200 === statusCode
+
+testRefreshTokenWrongSignature :: Opts -> Brig -> Http ()
+testRefreshTokenWrongSignature opts brig = do
+  uid <- userId <$> createUser "alice" brig
+  let redirectUrl = mkUrl "https://example.com"
+  let scopes = OAuthScopes $ Set.fromList [ReadSelf]
+  (cid, secret, code) <- generateOAuthClientAndAuthorizationCode brig uid scopes redirectUrl
+  let accessTokenRequest = OAuthAccessTokenRequest OAuthGrantTypeAuthorizationCode cid secret code redirectUrl
+  accessToken <- createOAuthAccessToken brig accessTokenRequest
+  key <- liftIO $ readJwk (fromMaybe "path to jwk not set" (Opt.setOAuthJwkKeyPair $ Opt.optSettings opts)) <&> fromMaybe (error "invalid key")
+  badRefreshToken <- liftIO $ do
+    claims <- verifyRefreshToken key (unOAuthToken $ oatRefreshToken accessToken)
+    OAuthToken <$> signRefreshToken badKey claims
+  let refreshAccessTokenRequest = OAuthRefreshAccessTokenRequest OAuthGrantTypeRefreshToken cid secret badRefreshToken
+  refreshOAuthAccessToken' brig refreshAccessTokenRequest !!! do
+    const 403 === statusCode
+    const "Forbidden" === statusMessage
+
+testRefreshTokenNoTokenId :: Opts -> Brig -> Http ()
+testRefreshTokenNoTokenId opts brig = do
+  uid <- userId <$> createUser "alice" brig
+  let redirectUrl = mkUrl "https://example.com"
+  let scopes = OAuthScopes $ Set.fromList [ReadSelf]
+  (cid, secret, _) <- generateOAuthClientAndAuthorizationCode brig uid scopes redirectUrl
+  key <- liftIO $ readJwk (fromMaybe "path to jwk not set" (Opt.setOAuthJwkKeyPair $ Opt.optSettings opts)) <&> fromMaybe (error "invalid key")
+  badRefreshToken <- liftIO $ OAuthToken <$> signRefreshToken key emptyClaimsSet
+  let refreshAccessTokenRequest = OAuthRefreshAccessTokenRequest OAuthGrantTypeRefreshToken cid secret badRefreshToken
+  refreshOAuthAccessToken' brig refreshAccessTokenRequest !!! do
+    const 403 === statusCode
+    const "Forbidden" === statusMessage
+
+testRefreshTokenNonExistingId :: Opts -> Brig -> Http ()
+testRefreshTokenNonExistingId opts brig = do
+  uid <- userId <$> createUser "alice" brig
+  let redirectUrl = mkUrl "https://example.com"
+  let scopes = OAuthScopes $ Set.fromList [ReadSelf]
+  (cid, secret, _) <- generateOAuthClientAndAuthorizationCode brig uid scopes redirectUrl
+  key <- liftIO $ readJwk (fromMaybe "path to jwk not set" (Opt.setOAuthJwkKeyPair $ Opt.optSettings opts)) <&> fromMaybe (error "invalid key")
+  badRefreshToken <-
+    liftIO $
+      OAuthToken <$> do
+        rid :: OAuthRefreshTokenId <- randomId
+        sub <- maybe (error "creating sub claim failed") pure $ idToText rid ^? stringOrUri
+        let claims = emptyClaimsSet & claimSub ?~ sub
```

Comment:
shouldn't this be a legit claim set with just one claim changed, rather than the empty claim set with one claim added?